### PR TITLE
Fixes #5225

### LINKED ISSUE #5225: compiler reject proc's param shadowing inside template
```nimrod
proc abc(x: int) =
  var x = x  #this one is ok
  
template something(name: untyped) =
  proc name(x: int) =
    var x = x #this one rejected by the compiler
    
something(what)
```

the error message:
```text
shadow.nim(6, 9) Error: cannot use symbol of kind 'param' as a 'var'
```

shadowing template's param could potentially cause problem, but shadowing proc's param should be ok.

### PR DESCRIPTION
**Pull request by loloiccl**

### CODE REVIEW

**Comment by Araq** [comment]
Shouldn't instead in the prepass a new scope be opened for parameters vs locals?

**Comment by loloiccl** [comment]
That looks much better.
I pushed new patch.

File: compiler/semtempl.nim [hunk 1]
```diff
@@ -263,9 +263,16 @@ proc semRoutineInTemplBody(c: var TemplCtx, n: PNode, k: TSymKind): PNode =
       n.sons[namePos] = ident
   else:
     n.sons[namePos] = semRoutineInTemplName(c, n.sons[namePos])
+  # onem scape for parameters
```
Comment (Araq) [inline compiler/semtempl.nim]: typos here.

Comment (loloiccl) [inline compiler/semtempl.nim]: sorry.
fixed.

**Pull request merged**